### PR TITLE
Support Redis auth

### PR DESCRIPTION
**Pull request by nebulade**

### CODE REVIEW

**Comment by d4l3k** [comment]
Thanks for taking the time to contribute to WebSync!

File: config/config.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
 "postgres": "postgres:@localhost/websync",
 "redis": {
   "host": null,
-  "port": null
+  "port": null,
+  "password": "null"
```
Comment (d4l3k) [inline config/config.json]: The default password should be `null`, not `"null"`.

**Comment by d4l3k** [comment]
If you make that one change I'd be happy to merge this.

File: lib/config.js [hunk 2]
```diff
@@ -15,6 +15,11 @@ config = JSON.parse(stripJsonComments(configLines));
 config.postgres = new pg.Client('tcp://' + config.postgres);
 config.postgres.connect();
 config.redis = _.extend(redis.createClient(config.redis.port, config.redis.host), config.redis);
+if (config.redis.password) {
+  config.redis.auth(config.redis.password, function (error) {
+    if (error) console.error('Redis auth failed:', error);
```
Comment (d4l3k) [inline lib/config.js]: You should add curly brackets to the if statements.

File: lib/config.js [hunk 3]
```diff
@@ -15,6 +15,13 @@ config = JSON.parse(stripJsonComments(configLines));
 config.postgres = new pg.Client('tcp://' + config.postgres);
 config.postgres.connect();
 config.redis = _.extend(redis.createClient(config.redis.port, config.redis.host), config.redis);
+if (config.redis.password) {
+  config.redis.auth(config.redis.password, function (error) {
+    if (error) {
+      console.error('Redis auth failed:', error);
```
Comment (nebulade) [inline lib/config.js]: That is some strange linter setup, why would it complain now with curly brackets, but not beforehand?

Comment (d4l3k) [inline lib/config.js]: I think it glitched earlier, but you can ignore that error.

**Comment by d4l3k** [comment]
It all looks good. Merged!

**Pull request merged**